services/api/src/api/routes/croissant.py

```diff
@@ -0,0 +1,238 @@
+import logging
+from collections.abc import Mapping
+from http import HTTPStatus
+from itertools import islice
+from typing import Any, Optional, TypedDict
+
+from datasets import Features, Image, Value
+from libapi.authentication import auth_check
+from libapi.exceptions import (
+    ApiError,
+    MissingRequiredParameterError,
+    UnexpectedApiError,
+)
+from libapi.request import get_request_parameter
+from libapi.utils import (
+    Endpoint,
+    are_valid_parameters,
+    get_cache_entry_from_steps,
+    get_json_api_error_response,
+    get_json_error_response,
+    get_json_ok_response,
+)
+from libcommon.processing_graph import ProcessingGraph
+from libcommon.prometheus import StepProfiler
+from starlette.requests import Request
+from starlette.responses import Response
+
+MAX_CONFIGS = 100
+
+
+class ParquetFile(TypedDict):
+    dataset: str
+    config: str
+    split: str
+    url: str
+    filename: str
+    size: int
+
+
+HF_TO_CRROISSANT_VALUE_TYPE = {
+    "string": "sc:Text",
+    "int32": "sc:Integer",
+    "int64": "sc:Integer",
+    "float32": "sc:Float",
+    "float64": "sc:Float",
+    "bool": "sc:Boolean",
+}
+
+
+def get_croissant_from_dataset_infos(dataset: str, infos: list[Mapping[str, Any]], partial: bool) -> Mapping[str, Any]:
+    distribution = [
+        {
+            "@type": "sc:FileObject",
+            "name": "repo",
+            "description": "The Hugging Face git repository.",
+            "contentUrl": f"https://huggingface.co/datasets/{dataset}",
+            "encodingFormat": "git+https",
+            "sha256": "https://github.com/mlcommons/croissant/issues/80",
+        }
+    ]
+    record_set = []
+    for info in infos:
+        config = info["config_name"]
+        features = Features.from_dict(info["features"])
+        fields: list[dict[str, Any]] = []
+        distribution_name = f"parquet-files-for-config-{config}"
+        distribution.append(
+            {
+                "@type": "sc:FileSet",
+                "name": distribution_name,
+                "containedIn": "repo",
+                "encodingFormat": "application/x-parquet",
+                "includes": f"{config}/*/*.parquet",
```

Comment:
it does not list the parquet files? What is this useful for? And we don't refer to the `refs/convert/parquet` branch? (sorry I don't know the Croissant format)

Reply:
Good catch, I forgot to add the branch x)

Currently no need to list the parquet files URLs, since mlcroissant supports globbing in git repos

Reply:
OK. And if we have the `config-info` in cache, it always means that we also have the parquet files, since it depends on `config-parquet-and-info`, right?

Reply:
(you might get both the parquet files URLs, and the info, from `config-parquet-and-info`, by the way

Reply:
> OK. And if we have the config-info in cache, it always means that we also have the parquet files, since it depends on config-parquet-and-info, right?

yes exactly

> (you might get both the parquet files URLs, and the info, from config-parquet-and-info, by the way)

I used "dataset-info" to be able to do one single call to the db

Reply:
yes, good idea, in particular if we don't need the detailed list of config parquet files